components/scream/scripts/test-all-scream

```diff
@@ -41,38 +41,33 @@ OR
 
     parser.add_argument("-s", "--submit", action="store_true", help="Submit results to dashboad, requires machine")
 
-    parser.add_argument("-p", "--parallel", action="store_true", help="Launch the different build types stacks in parallel")
+    parser.add_argument("-p", "--parallel", action="store_true",
+                        help="Launch the different build types stacks in parallel")
 
-    parser.add_argument("-f", "--fast-fail", action="store_true", help="Stop testing when the first failure is detected")
+    parser.add_argument("-f", "--fast-fail", action="store_true",
+                        help="Stop testing when the first failure is detected")
 
-    default_baseline_ref = "HEAD" if "JENKINS_HOME" in os.environ else "origin/master"
-    parser.add_argument("-b", "--baseline-ref", default=default_baseline_ref,
-                        help="What commit to use to generate baselines")
-    default_baseline_dir = "NONE"
-    parser.add_argument("--baseline-dir",default=default_baseline_dir,
-                        help="Use baselines from the given directory, skip baseline creation. Ignores -b/--baseline options")
+    parser.add_argument("-b", "--baseline-ref", default=None, # default will be computed later
+        help="What commit to use to generate baselines. Default is merge-base of current commit and origin/master (or HEAD if --keep-tree)")
 
-    parser.add_argument("-m", "--machine", help="Provide machine name. This is required for internal kokkos builds and dashboard submission")
+    parser.add_argument("--baseline-dir", default=None,
+        help="Use baselines from the given directory, skip baseline creation.")
+
+    parser.add_argument("-m", "--machine",
+        help="Provide machine name. This is required for internal kokkos builds and dashboard submission")
 
     parser.add_argument("--no-tests", action="store_true", help="Only build baselines, skip testing phase")
 
-    parser.add_argument("--keep-tree", action="store_true", help="Allow to keep the current work tree when testing against HEAD (only valid with `-b HEAD`)")
+    parser.add_argument("-d", "--keep-tree", action="store_true",
```

Comment:
I find it confusing when the single-letter flag is not the initial of the verbose flag. In this case, you cannot use `-k`, since that's already the short for `--kokkos`. I would do away with single-letter flag here. Besides, `-d` sounds like "delete" (although I'm assuming you meant 'dirty').